### PR TITLE
Dto

### PR DESCRIPTION
**Pull request by Volodymyr-Mykychak**

### CODE REVIEW

File: src/main/java/com/store/book/repository/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+@Repository
+public interface BookRepository {
+    Book save(Book book);
+
+    Optional<Book> findById(Long id);
+
+    List<Book> findAll();
+
+    List<Book> findAllByTitle(String title);
+
```
Comment (kozhukhovsky) [inline src/main/java/com/store/book/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/com/store/book/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw e;
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```
Comment (kozhukhovsky) [inline src/main/java/com/store/book/repository/impl/BookRepositoryImpl.java:38]: catch?

File: src/main/java/com/store/book/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```
Comment (kozhukhovsky) [inline src/main/java/com/store/book/repository/impl/BookRepositoryImpl.java:48]: catch?

File: src/main/java/com/store/book/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            return entityManager.createQuery("SELECT e FROM Book e", Book.class).getResultList();
+        }
+    }
+
+    @Override
+    public List<Book> findAllByTitle(String title) {
+        String lowerCaseName = title.toLowerCase();
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```
Comment (kozhukhovsky) [inline src/main/java/com/store/book/repository/impl/BookRepositoryImpl.java:58]: catch?

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/com/store/book/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> findAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("{id}")
```
Comment (olekskov) [inline src/main/java/com/store/book/controller/BookController.java:27]:
```suggestion
    @GetMapping("/{id}")
```

File: src/main/java/com/store/book/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("" + (long) (Math.random() * 9_000_000_000_000L) + 1_000_000_000_000L);
```
Comment (olekskov) [inline src/main/java/com/store/book/service/impl/BookServiceImpl.java:23]: It seems logic of setting Isbn will be more appropriate in mapper to obtain ready to use model

**Review by olekskov — APPROVED** [review]

**Pull request merged**